### ISSUE TITLE
How hard would it be to rework boson to support Ruby 1.8.7?

### ISSUE THREAD

**Issue by bgupta** [cldwalker/boson#38]
I ask because I want to make any CLI tools I write as widely distributable as possible, and most distros still ship with 1.8.7.

Thanks,
Brian

**Comment by cldwalker** [comment]
boson uses Method#parameters to determine args of a command. Before I.9 I depended on ugly, brute force hacks. I'm sure boson depends on 1.9 in a few more places. For 1.8.7, you're welcome to still use boson < 1.0 but I'm only interested in 1.9 for boson. If you haven't already read, [1.8.7 is on it's way out](http://www.ruby-lang.org/en/news/2011/10/06/plans-for-1-8-7/)

**Comment by bgupta** [comment]
I understand the issues with Ruby 1.8.7 being on it's way out,
unfortunately pretty much every major distro is still shipping 1.8.7
as the default Ruby. (Even Ubuntu 12.04 LTS, which ships in a couple
of days, is still going to ship with 1.8.7, and requiring folks to be
super aware of RVM and what not doesn't make sense for my use case.
IE: I am going to need to distribute this tool via OS packages).

That said, I understand your position, but I was hoping it might not
be a lot of work, since boson 1.x doesn't support the named command
with nested subcommands style usage I was looking for. (IE: Being able
to write a git-style command suite.)

That said, I am looking at clamp, and gli-2 which seem to support most
of what I am looking for, and one of which will hopefully meet our
needs. (Thor doesn't seem to support global options, so it's out.)

I evaluated the cli framework space a couple of months ago, and Boson
seemed the most feature-complete, but the 1.8.7 requirement at this
point is a bit of a show stopper. I imagine that while I am looking at
1.8.7 for now, I will end up having to support both 1.8.7 and 1.9 at
some point in the relatively near future. (Maybe in 12 months or so.)
I can't see being 1.9 only for at least a year or two years,
considering all the OSes that Foreman needs to support. (We just
earlier this year dropped Ruby 1.8.5 support, largely because we
migrated to Rails3 and Rails3 dropped support.)

I am documenting here, one of the more important (to me anyway) CLI
tools I am struggling to get started writing (struggling finalizing on
a CLI framework.): http://theforeman.org/projects/foreman/wiki/Hammer
(It is a replacement for a much simpler tool that I wrote that didn't
use subcommands).

Basically, it's a commandline tool for working with Foreman, which is
a server provisioning dashboard, that integrated with puppet. (Foreman
has a fairly robust REST API, in that everything in the GUI, is
accesible via the API, so developing a CLI client, is a nature fit.)

Thanks for your response.

-Brian

On Mon, Apr 23, 2012 at 4:24 AM, Gabriel Horner
reply@reply.github.com
wrote:

**Comment by cldwalker** [comment]
I understand your needs for 1.8.7 but they don't overlap with mine. Best of luck. A few more command frameworks to check out ;):
- http://github.com/jm/sweep
- http://github.com/bmizerany/frylock
- http://github.com/dakrone/nsm-console
- http://simpleconsole.rubyforge.org/index.html
- http://github.com/delano/drydock
- https://github.com/postmodern/rprogram
- http://github.com/hipe/hipe-cli
- http://github.com/masa16/pwrake
- http://github.com/botanicus/nake
- http://github.com/koraktor/rubikon
- http://github.com/markryall/shell_shock
- http://github.com/edelbalso/rcli
- http://tap.rubyforge.org/
- http://github.com/visionmedia/commander
- http://github.com/jashmenn/git-style-binaries
- https://github.com/blambeau/quickl
- http://clio.rubyforge.org/
- http://github.com/quix/rake
- http://github.com/jugyo/cinatra
- http://github.com/postmodern/sake/tree/master/
- http://github.com/karottenreibe/caty
- http://github.com/justinbaker/rtask
- http://github.com/ahoward/main
